openedx/features/badging/helpers.py

```diff
@@ -116,3 +120,32 @@ def filter_earned_badge_by_joined_team(user, course, earned_badges):
         earned_badge for earned_badge in earned_badges if
         earned_badge.community_id == course_team[TEAM_ROOM_ID_KEY]
     ]
+
+
+def send_user_badge_notification(user, my_badge_url, badge_name):
+    """
+    Send user new badge notification
+    :param user: Notification sender user
+    :param my_badge_url: Redirect url to my_badge view on notification click
+    :param badge_name: Newly earned badge
+    """
+    context = {
+        'badge_name': badge_name
+    }
+
+    body_short = render_to_string('philu_notifications/templates/user_badge_earned.html', context)
+
+    message_type = get_notification_type(
+        NOTIFICATION_TYPE.format(prefix='philu.badging', type='user-badge-earned')
+    )
+
+    message = NotificationMessage(
+        msg_type=message_type,
+        payload={
+            'from_user': user.username,
```

Comment:
do we really need this key in payload? 
Please check it without passing `from_user`
because logically it doesn't seem correct as this notification is not like chat notifications. This notification is sent from platform not from any user.

Reply:
as per current notification format this is how it appears with user initials on left side.If we are going to change that then it will have effect both on platform side and on community which is not within the scope of this story.

Reply:
👍